### PR TITLE
Jh/modulecard tags

### PR DESCRIPTION
**Pull request by jhenry09**
This is a draft for the tags section on module cards. the my modules and explore page were tested. Let me know about the style. If the tags overflow even when expanded, there is a scroll bar.
modulecard-tags

### CODE REVIEW

**Comment by Cliftonz** [comment]
I think this is good but I think would should make some changes.

I think that the tags should have 3 states instead of two.
The first is compressed similar to how Trello works with their tags. Only small bubbles no words.

The second is open like you have currently where you see the bubble and the word. However, I think the expansion button should be a ... After the last tag instead of a big eye catchy arrow.
The last is expanded just like you have now where you can see all the tags. I do think the arrow on the top right should be replaced with an X.

The last detail is make sure you test this with a module that has no tags.

**Comment by Cliftonz** [comment]
Also, how did you get to adding 25,000 lines of code?

I did you merge this branch with another?

**Comment by jhenry09** [comment]
> I think this is good but I think would should make some changes.
> 
> I think that the tags should have 3 states instead of two. The first is compressed similar to how Trello works with their tags. Only small bubbles no words.  The second is open like you have currently where you see the bubble and the word. However, I think the expansion button should be a ... After the last tag instead of a big eye catchy arrow. The last is expanded just like you have now where you can see all the tags. I do think the arrow on the top right should be replaced with an X.
> 
> The last detail is make sure you test this with a module that has no tags.

Got it. I will make these changes.

**Comment by jhenry09** [comment]
> Also, how did you get to adding 25,000 lines of code?
> 
> I did you merge this branch with another?

I did merge this with my tag editor branch but I am not sure why package-lock.json had 24k additions. Let me take a look at this when I get home this afternoon.

**Comment by Cliftonz** [comment]
> 
> 
> > Also, how did you get to adding 25,000 lines of code?
> > I did you merge this branch with another?
> 
> I did merge this with my tag editor branch but I am not sure why package-lock.json had 24k additions. Let me take a look at this when I get home this afternoon.

Make sure that you merge the editor in first so you do not get merge conflicts.

**Comment by jasekiw** [comment]
@jhenry09 Can you resolve the merge conflicts in this branch? Somehow you have merge conflicts with yourself

**Comment by jhenry09** [comment]
435_gif2
Updated styling

**Comment by Cliftonz** [comment]
discription text should be smaller and grey

**Comment by jhenry09** [comment]
> discription text should be smaller and grey

image
better?

File: src/components/ModuleCard/ModuleCard.tsx [hunk 1]
```diff
@@ -8,34 +8,71 @@ import {isAuthenticated} from '../../redux/selectors/entities';
...
   buttonLink?: string;
   buttonAction?: () => void;
 }
+interface ModuleCardState {
+  viewTagsExpanded: boolean;
+}
+
+function doNothing() {
+  // do nothing
+}
```
Comment (jasekiw) [inline src/components/ModuleCard/ModuleCard.tsx]: this probably would be better located in the `util/index.ts` file

File: src/pages/PublicModule/PublicModule.tsx [hunk 2]
```diff
@@ -12,13 +12,18 @@ import {Alert, Button, Card} from 'react-bootstrap';
...
 import {LoadingButton} from '../../util/LoadingButton';
+import {TagEditor} from '../../components/TagEditor/TagEditor';
 
 interface MyModuleState {
   module: Module;
   message?: string;
   startingModule: boolean;
 }
 
+function doNothing() {
```
Comment (jasekiw) [inline src/pages/PublicModule/PublicModule.tsx]: reuse the same doNothing method that you will move to the util folder.

File: src/components/TagEditor/TagEditor.tsx [hunk 3]
```diff
@@ -20,6 +20,10 @@ export function TagEditor({tags, tagSuggestions, onInput, onAdd, onDelete, mes,
       setClassName('enabled success');
     else if (mes === 'success' && !editing)
       setClassName('disabled success');
+    else if (mes === 'display-only')
```
Comment (jasekiw) [inline src/components/TagEditor/TagEditor.tsx]: No need to have useEffect here, just assign the className directly.

File: src/components/ModuleCard/ModuleCard.tsx [hunk 4]
```diff
@@ -8,34 +8,71 @@ import {isAuthenticated} from '../../redux/selectors/entities';
...
-            {module.description.substring(0, 150)}
-          </Card.Text>
-        </Card.Body>
-        <Card.Footer style={{display: 'flex', justifyContent: 'space-between', alignItems: 'center'}}>
-          <small className='text-muted'>{getLocalDateTimeString(module.updatedAt)}</small>
-          {this.getStartButton()}
-        </Card.Footer>
-      </Card>
+        <Card className={Styles.card}>
+          {/*<Card.Img variant='top' src={TestImage}/>*/}
```
Comment (jasekiw) [inline src/components/ModuleCard/ModuleCard.tsx]: Remove this commented out code

**Review by jasekiw — CHANGES_REQUESTED** [review]

**Pull request merged**